src/com/aca/ArmineArzumanyan/homeWork_1/items/Item.java

```diff
@@ -0,0 +1,44 @@
+package com.aca.ArmineArzumanyan.homeWork_1.items;
+
+public abstract class Item {
+    private String name;
+    private final int code;
+    private int price;
+    private int count;
+
+    public Item(String name, int code, int count, int price) {
+        this.code = code;
```

Comment:
ok you choose this way to achieve the abstraction but you should think that the user of this class can pass duplicate codes to items or names, are you sure that all will be ok in that case?

Reply:
is it more convenient  to use HashMap<>  in place of List<>?